### ISSUE TITLE
Enhance profile search: prevent users from selecting redundant custom cause areas or countries

### ISSUE THREAD

**Issue by victor-yunenko** [rtcharity/eahub.org#978]
The search wireframe at the moment looks as following:

![image](https://user-images.githubusercontent.com/36416236/93722614-c5cb5e00-fba0-11ea-8c26-7c2fdd6e0432.png)

An interesting issue is that every one of those dropdown fields is a text field for the users - ie everyone can input anything in the `country` field, instead of just selecting `US` from a dropdown. As a consequence once we implement the dropdown inside of our `country` select we're going to see something as:
- UK
- United Kingdom
- UK/Poland
- UK/USA
- Ukraine
- Union of Soviet Socialist Republics
- United States
- US
- USA
- USA / Belgium

Especially I'm a fan of that `Union of Soviet Socialist Republics` - and I'm not making it up - I looked up the users countries in the database 😁 

During the PoC I'm going to ignore it, although at some point we will have to clean up our ~5k user profiles and prevent users from inputting random text in those fields.

As an option for the countries & cities - we can ignore their input and derive the location based on the attached latitude and longitude.

Ideally I would like to see a system similar to stackexchange - where users are encouraged to reuse the existing tags instead of creating their own:

![image](https://user-images.githubusercontent.com/36416236/93722641-1cd13300-fba1-11ea-8fcd-b1eab9f4fc9d.png)

**Comment by vagarwalla** [comment]
I wasn't aware that countries were open text. Here are my thoughts:

**Use case of multiple locations**
Some people are from 1 country/city, but living in another and want to be found by people intersted in both areas.

**For a proper solution**
- There is only 1 country & 1 city field but both are multi-select. Countries should be multi-select dropdown (no extra options). Cities can use the tagging system like you suggest where users can create new options if needed (useful for smaller cities/towns). In this version it may be possible to get the weird city/country combos that Seb mentioned #972. 
- Geotagging occurs by the first country selected

Questions for you/@sebgrebe
- How much time would this take? 
- My gut feeling is that we should add this to MVP plans because it seems pretty important, but I'm quite worried about scope creep and the tagging system seems like it would take some time to implement.
- So i've suggested some MVPs workarounds:

**Quick brainstorm:**
- In excel you can do a "select all" for a partial match on words. Can we do something similar? So at least 80% of US or UK are selected (ignoring the e.g. "United States" entries)
![image](https://user-images.githubusercontent.com/32846756/93726200-39905980-fbe7-11ea-8bee-cf7d4800281e.png)
- Or 2, We clean up the data for just the US/UK since they are the most common countries where we choose. This will work for a little while, but if we get a lot of new users they will again make the data messy. So it's not a great solution

**Comment by victor-yunenko** [comment]
> My gut feeling is that we should add this to MVP plans because it seems pretty important, but I'm quite worried about scope creep and the tagging system seems like it would take some time to implement.

I agree with the feature creep issue - I naturally wouldn't include it in the first MVP since it would definitely require firstly implementing the initial version working version, and then - instead of rolling it out to the live server - we would need ~1 more month of internal work to handle those selection, tags and multiple cities/countries per users problems. While if we push it live first - the end users might immediately point out to a set of far more pressing issues :)

While personally I like the idea of multiple cities/countries 👍

> In excel you can do a "select all" for a partial match on words. Can we do something similar? So at least 80% of US or UK are selected (ignoring the e.g. "United States" entries)

Regarding `select all` button - not sure, might require too much effort. But the ability to select several countries/cities at the same time - of course, I would implement it like that in any case 👌

**Comment by vagarwalla** [comment]
Just to recap, this issue will be addressed post-MVP launch:
- Multi-city and multi-country selection (using a tagging system)
- select all filter button for the search
